In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import scipy.stats as stats

import statsmodels.api as sm
from statsmodels.tsa.filters.hp_filter import hpfilter
from scipy.stats import norm
from scipy import stats


# Ejercicio 1

En una regresión lineal simple con intercepto:

    y_i = alpha + beta x_i + u_i

el estimador MCO de la pendiente beta mide cuánto cambia y, en promedio, ante
un cambio unitario en x.

La fórmula del estimador de la pendiente es:

    beta_hat = sum((x_i - x_bar)(y_i - y_bar)) / sum((x_i - x_bar)^2)

El numerador mide cómo se mueven conjuntamente x e y respecto de sus medias,
es decir, la covarianza muestral multiplicada por n-1:

    Cov(x,y) = sum((x_i - x_bar)(y_i - y_bar)) / (n-1)

El denominador mide la dispersión de x respecto de su media, es decir, la
varianza muestral de x multiplicada por n-1:

    Var(x) = sum((x_i - x_bar)^2) / (n-1)

Entonces:

    beta_hat =
    [sum((x_i - x_bar)(y_i - y_bar)) / (n-1)]
    ------------------------------------------------
    [sum((x_i - x_bar)^2) / (n-1)]

Como el factor (n-1) aparece tanto arriba como abajo, se cancela. Por lo tanto:

    beta_hat = Cov(x,y) / Var(x)

Aplicado a finanzas, si y es la prima de riesgo de un activo y x es la prima de
riesgo del mercado, el beta del activo es la covarianza muestral entre la prima
de riesgo del activo y la prima de riesgo del mercado, dividida por la varianza
muestral de la prima de riesgo del mercado.

Por eso, el beta no es solamente una pendiente de regresión: también puede
interpretarse como una covarianza normalizada por el riesgo del mercado.


# Ejercicio 2

a) 

La pendiente estimada es 0,69758673. Esto significa que, por cada aumento de una
unidad en la prima de riesgo del mercado, es decir, en el rendimiento del S&P 500
por encima de la tasa libre de riesgo, la prima de riesgo de IBM aumenta en
promedio 0,69758673 unidades.

Desde el punto de vista financiero, este coeficiente es el beta de IBM respecto
del mercado. Como es menor que 1, IBM tiene una sensibilidad menor que la del
mercado: tiende a moverse en la misma dirección que el mercado, pero con menor
intensidad.

Por ejemplo, si la prima de riesgo del mercado aumenta 1 punto porcentual, la
prima de riesgo esperada de IBM aumenta aproximadamente 0,6976 puntos
porcentuales.

B) 

Para testear si la pendiente es estadísticamente distinta de cero planteo:

H0: beta = 0
HA: beta != 0

Uso el p-value asociado al estadístico t de la pendiente. Según la salida de
Excel, el p-value es 1.8687E-05, que es menor al nivel de significancia del 5%.

Por lo tanto, rechazo la hipótesis nula de que beta = 0. Esto indica que la
prima de riesgo del mercado tiene un efecto estadísticamente significativo sobre
la prima de riesgo de IBM.

C)

Para testear si beta es igual a 1, planteo:

H0: beta = 1

HA: beta != 1

El p-value de Excel no sirve directamente en este caso, porque Excel reporta el
test contra beta = 0. Por eso calculo manualmente el estadístico:

t = (beta_hat - 1) / SE(beta_hat)

t = (0.69758673 - 1) / 0.14933314 = -2.0251

gl = n - cantidad de parámetros estimados

gl = 59 - 2

gl = 57


Con 57 grados de libertad, el p-value a dos colas es aproximadamente 0.0475.

Como el p-value es menor que 0.05, rechazo H0 al 5%. Por lo tanto, hay evidencia
estadística para decir que el beta de IBM es distinto de 1. Además, como el beta
estimado es menor que 1, IBM parece tener una sensibilidad al mercado menor que
la cartera de mercado.

In [3]:
beta_hat = 0.69758673
se_beta = 0.14933314
beta_0 = 1

t_stat = (beta_hat - beta_0) / se_beta
t_stat

-2.0250914833773663

In [4]:
gl = 57

p_value = 2 * (1 - stats.t.cdf(abs(t_stat), df=gl))

t_stat, p_value

(-2.0250914833773663, np.float64(0.047550585964967196))

D)

El intervalo de confianza del 95% para el parámetro beta está dado por los

valores reportados en la salida de Excel:

IC 95% = [0.39855229 ; 0.99662118]


Esto significa que, con un 95% de confianza, el verdadero beta de IBM se
encuentra entre 0.3986 y 0.9966. Como el intervalo no contiene al valor 0,
esto es consistente con rechazar la hipótesis nula beta=0.

 Además, el intervalo
queda levemente por debajo de 1, lo cual es consistente con que IBM tenga una
sensibilidad menor que la del mercado.

In [7]:
beta_hat = 0.69758673
se_beta = 0.14933314
gl = 57

t_critico = stats.t.ppf(0.975, df=gl)

li = beta_hat - t_critico * se_beta
ls = beta_hat + t_critico * se_beta

li, ls

(np.float64(0.3985522752225318), np.float64(0.9966211847774683))

beta_hat ± t_critico * SE(beta_hat)

e) 

Para evaluar si el intercepto es estadísticamente distinto de cero planteo:

H0: alpha = 0
HA: alpha != 0

Según la salida de Excel, el p-value asociado al intercepto es 0.6950. Como este
valor es mayor que el nivel de significancia del 5%, no rechazo la hipótesis
nula.

Por lo tanto, no hay evidencia estadística suficiente para afirmar que el
intercepto sea distinto de cero. 

En términos financieros, el alpha de IBM no parece ser significativamente diferente de cero una vez controlado por la prima
de riesgo del mercado.

f)

Sí, a partir de la salida de Excel puedo obtener la correlación entre la prima
de riesgo de IBM y la prima de riesgo del mercado.

En una regresión simple con intercepto se cumple que el R² es igual al cuadrado
de la correlación muestral entre la variable dependiente y la variable
explicativa:

R² = corr(x,y)^2

La salida de Excel reporta un R² de 0.2768468. Por lo tanto:

corr(x,y) = sqrt(0.2768468) = 0.52616233

Además, como la pendiente estimada es positiva, la correlación también debe ser
positiva.

Entonces, la correlación entre la prima de riesgo de IBM y la prima de riesgo
del mercado es aproximadamente 0.5262.

# Ejercicio 3

In [15]:
ruta = "/Users/bautistagianfrancisco/Library/Mobile Documents/com~apple~CloudDocs/MIA 2026/2 Trim/Modelado Estocastico/Bases de Datos MIA103/MIA103 Ejercitación 3 Datos.xlsx"

ceo = pd.read_excel(
    ruta,
    sheet_name="Datos_Ejer_3_CEO"
)

ceo.head()

,Ganancias ($millones),Compensación al CEO ($millones)
0,357.0,0.7
1,48.0,0.7
2,932.0,0.8
3,366.0,0.7
4,83.0,0.8


In [16]:
ceo.columns

Index(['Ganancias ($millones)', 'Compensación al CEO ($millones)'], dtype='object')

In [22]:
ceo.columns = ["profits", "salary"]

ceo.head()

,profits,salary
0,357.0,0.7
1,48.0,0.7
2,932.0,0.8
3,366.0,0.7
4,83.0,0.8


In [23]:

y = ceo["salary"]      # compensacion CEO
x = ceo["profits"]     # ganancia empresa

x = sm.add_constant(x)

modelo_ceo = sm.OLS(y, x).fit()

print(modelo_ceo.summary())

                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.434
Model:                            OLS   Adj. R-squared:                  0.426
Method:                 Least Squares   F-statistic:                     52.24
Date:                Wed, 26 Aug 2026   Prob (F-statistic):           5.50e-10
Time:                        21:20:28   Log-Likelihood:                -73.655
No. Observations:                  70   AIC:                             151.3
Df Residuals:                      68   BIC:                             155.8
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.6000      0.112      5.342      0.0

La pendiente estimada es aproximadamente 0.0008. Esto significa que, por cada
aumento de 1 millón de dólares en las ganancias de la empresa, la compensación
esperada del CEO aumenta en promedio 0.0008 millones de dólares.

Como la compensación también está medida en millones de dólares, 0.0008 millones
equivalen aproximadamente a 800 dólares.

Por lo tanto, un aumento de 1 millón de dólares en las ganancias de la empresa
se asocia con un aumento promedio de alrededor de 800 dólares en la compensación
del CEO.

In [26]:
beta_hat = modelo_ceo.params["profits"]
t_stat = modelo_ceo.tvalues["profits"]
p_value = modelo_ceo.pvalues["profits"]

print("Beta estimado:", beta_hat)
print("t-stat:", t_stat)
print("p-value:", p_value)

Beta estimado: 0.0008423268484880064
t-stat: 7.227937463627728
p-value: 5.503310411082839e-10


In [25]:
alpha = 0.01

p_value = modelo_ceo.pvalues["profits"]

if p_value < alpha:
    print("Rechazo H0: la pendiente es estadísticamente distinta de cero al 1%.")
else:
    print("No rechazo H0: no hay evidencia suficiente al 1%.")

Rechazo H0: la pendiente es estadísticamente distinta de cero al 1%.


Para testear si las ganancias de la empresa tienen una relación estadísticamente
significativa con la compensación del CEO, planteo:

H0: beta = 0

HA: beta != 0

donde beta es la pendiente asociada a profits.

Según la regresión, el estadístico t de la pendiente es aproximadamente 7.228 y
el p-value es aproximadamente 0.000, menor al nivel de significancia del 1%.

Por lo tanto, rechazo la hipótesis nula. Hay evidencia estadística suficiente
para afirmar que, en esta muestra, mayores ganancias de la empresa se asocian
con mayor compensación al CEO.

In [27]:
r2 = modelo_ceo.rsquared

print("R²:", r2)
print("Porcentaje explicado:", r2 * 100)

R²: 0.43447889049101684
Porcentaje explicado: 43.44788904910168


El porcentaje de la variabilidad en las compensaciones a los CEOs explicado por
las ganancias de las empresas está dado por el R² de la regresión.

En este caso, el R² es aproximadamente 0.434. Esto significa que alrededor del
43.4% de la variabilidad observada en la compensación de los CEOs puede ser
explicada por las ganancias de las empresas.

El porcentaje restante, aproximadamente 56.6%, queda explicado por otros factores
que no están incluidos en esta regresión simple y por el término de error.